homeassistant/components/fritz/update.py

```diff
@@ -0,0 +1,84 @@
+"""Support for AVM FRITZ!Box update platform."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any, Final
+
+from homeassistant.components.update import (
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .common import AvmWrapper, FritzBoxBaseEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class FritzUpdateEntityEntityDescription(UpdateEntityDescription):
+    """Describes AVM FRITZ!Box update entity."""
+
+
+UPDATE_ENTITIES: Final = [
+    FritzUpdateEntityEntityDescription(
+        key="update",
+        name="Firmware Update",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    )
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up AVM FRITZ!Box update entities."""
+    _LOGGER.debug("Setting up AVM FRITZ!Box update entities")
+    avm_wrapper: AvmWrapper = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        FritzBoxUpdateEntity(avm_wrapper, entry.title, description)
+        for description in UPDATE_ENTITIES
+    )
+
+
+class FritzBoxUpdateEntity(FritzBoxBaseEntity, UpdateEntity):
+    """Mixin for update entity specific attributes."""
+
+    def __init__(
+        self,
+        avm_wrapper: AvmWrapper,
+        device_friendly_name: str,
+        description: FritzUpdateEntityEntityDescription,
+    ) -> None:
+        """Init FRITZ!Box connectivity class."""
+        self.entity_description = description
+        self._attr_name = f"{device_friendly_name} {description.name}"
+        self._attr_unique_id = f"{avm_wrapper.unique_id}-{description.key}"
+        super().__init__(avm_wrapper, device_friendly_name)
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+        return self._avm_wrapper.current_firmware
```

Comment:
So, you are using entity descriptions, but this method is hardcoded. That is a bit weird, as it takes away the flexibility the entity descriptions bring.

Are more update entities expected from this integration in the future?

If so, then this should be part of the entity description. If not, consider dropping the entity description instead.